decompilers/d2d_ghidra/src/main/java/decomp2dbg/D2DGhidraServerAPI.java

```diff
@@ -152,5 +152,54 @@ public Map<String, Object> global_vars() {
 		this.server.plugin.gVarCache = resp;
 		return resp;
 	}
+
+	public Map<String, Object> structs() {
+		// check the cache before doing hard work!
+		if(!this.server.plugin.structCache.isEmpty())
+			return this.server.plugin.structCache;
 	
+		// if we are here, this is first connection!
+		ArrayList<Object> structInfos = new ArrayList<>();
+		var program = this.server.plugin.getCurrentProgram();
+		var dtm = program.getDataTypeManager();
+		var structs = dtm.getAllStructures();
+
+		while (structs.hasNext()) {
+			var struct = structs.next();
+			Map<String, Object> structInfo = new HashMap<>();
+			structInfo.put("name", struct.getName());
+			// Enumerate members
+			var members = struct.getComponents();
+			// For unknown reasons, the API claims that some targets have structures with a crazy number of members (in the millions).
+			// This causes an infinite loop.
+			// May be caused by a bug in some other plugin, not sure.
+			if (members.length > 1000) {
+				continue;
+			}
+			ArrayList<Object> memberInfo = new ArrayList<>();
+			var unnamedMembers = 0;
+			for (var member : members) {
+				Map<String, Object> memberData = new HashMap<>();
```

Comment:
Although the generic `.structs()` API did not define the need for an `offset` per variable, I think it could be beneficial for future work if you gave what offset each of these members is inside their struct. It's optional, and if you feel it's too hard you can just resolve this comment.

Reply:
Implemented.